iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -481,7 +484,19 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
           .map(x -> x.getJobConf().get(ConfVars.REWRITE_POLICY.varname))
           .orElse(RewritePolicy.DEFAULT.name()));
 
-      commitOverwrite(table, branchName, startTime, filesForCommit, rewritePolicy);
+      Integer compactionSpecId = outputTable.jobContexts.stream()
+          .findAny()
+          .map(x -> x.getJobConf().get(CompactorContext.COMPACTION_SPEC_ID))
+          .map(Integer::valueOf)
+          .orElse(null);
+
+      String compactionPartitionPath = outputTable.jobContexts.stream()
+          .findAny()
+          .map(x -> x.getJobConf().get(CompactorContext.COMPACTION_PARTITION_PATH))
+          .orElse(null);
+
+      commitOverwrite(table, branchName, startTime, filesForCommit, rewritePolicy, compactionSpecId,
```

Comment:
should we handle compaction separately, in diff method?

Reply:
I was also thinking about it; Done.